cfy_manager/components/restservice/db.py

```diff
@@ -44,63 +43,104 @@
 REST_HOME_DIR = '/opt/manager'
+STAGE_DB_NAME = 'stage'
+COMPOSER_DB_NAME = 'composer'
 
 
-def _connect_to_db(query_func):
-    def wrapper(*args, **kwargs):
-        pg_config = config[POSTGRESQL_CLIENT]
-        db_connection_string = \
-            'postgres://{user}:{password}@{hostname_and_port}/{db}'.format(
-                user=pg_config['username'],
-                password=pg_config['password'],
-                hostname_and_port=pg_config['host'],
-                db=pg_config['db_name']
-            )
-        try:
-            connection = create_engine(db_connection_string,
-                                       poolclass=NullPool).connect()
-
-            return_value = query_func(connection=connection, *args, **kwargs)
-
-            connection.close()
-            return return_value
-        except Exception as err:
-            logger.debug('{0} - Database not initialized yet, proceeding...'
-                         .format(err))
-            return constants.DB_NOT_INITIALIZED
-    return wrapper
+@contextmanager
+def _connect_to_db(cloudify_db=False):
+    pg_config = config[POSTGRESQL_CLIENT]
+    db_to_use = pg_config['db_name'] if cloudify_db \
+        else pg_config['server_db_name']
+    db_connection_string = \
+        'postgres://{user}:{password}@{hostname_and_port}/{db}'.format(
+            user=pg_config['server_username'],
+            password=pg_config['server_password'],
+            hostname_and_port=pg_config['host'],
+            db=db_to_use
+        )
+    # DROP and CREATE db queries are not permitted during a transaction
+    # so we AUTOCOMMIT
+    connection = create_engine(db_connection_string,
+                               poolclass=NullPool,
+                               isolation_level="AUTOCOMMIT").connect()
+    yield connection
+    connection.close()
 
 
 def prepare_db():
-    logger.notice('Configuring SQL DB...')
     pg_config = config[POSTGRESQL_CLIENT]
+    cloudify_db_name = pg_config['db_name']
+    server_username = pg_config['server_username'].split('@')[0]
+    username = pg_config['username'].split('@')[0]
+    password = pg_config['password']
+
+    drop_database_query = "DROP DATABASE IF EXISTS {db_name};"
+    drop_user_query = "DROP USER IF EXISTS {username};"
+    create_user_query = "CREATE USER {username} WITH PASSWORD '{password}';"
+    grant_role_query = "GRANT {username} TO {pg_server_username};"
+    create_db_query = "CREATE DATABASE {db_name};"
+    grant_all_privileges_query = \
+        "GRANT ALL PRIVILEGES ON DATABASE {db_name} TO {username};"
+    alter_user_query = "ALTER USER {username} CREATEDB;"
+    alter_db_owner_query = "ALTER DATABASE {db_name} OWNER TO {username};"
+    revoke_role_query = "REVOKE {username} FROM {pg_server_username};"
+
+    logger.notice('Configuring SQL DB...')
+    with _connect_to_db() as connection:
+        logger.debug('Cleaning server of old Cloudify data')
```

Comment:
Dropping all existing old Cloudify DB data
Or something in this spirit,because the cleaning server is a little strange wording